server/src/main/java/com/exacaster/lighter/application/sessions/SessionService.java

```diff
@@ -60,10 +66,26 @@ public Application createSession(String sessionId, SessionParams sessionParams)
     }
 
     public Application createPermanentSession(String sessionId, SubmitParams params) {
+        validateSessionLimit(ApplicationType.PERMANENT_SESSION);
         return createSession(params, sessionId, ApplicationType.PERMANENT_SESSION);
     }
 
+    private void validateSessionLimit(ApplicationType type) {
+        if (!SESSIONS.contains(type)) {
+            return;
+        }
+
+        var allSessions = applicationStorage.findApplications(SESSIONS, 0, Integer.MAX_VALUE).size();
+        var maxRunning = appConfiguration.getMaxRunningSessions();
+        if (maxRunning != null && allSessions >= maxRunning) {
```

Comment:
Can we move `maxRunning != null` check higher, before `findApplications` call?
That way we would avoid querying database, in cases when `MaxRunningSessions` feature is not used/not configured.